rcode-reader/src/main/java/eu/rssw/pct/RCodeInfo.java

```diff
@@ -0,0 +1,723 @@
+ * contact AT riverside DASH software DOT fr
+ * 
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package eu.rssw.pct;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+
+import eu.rssw.pct.elements.BufferElement;
+import eu.rssw.pct.elements.DataSourceElement;
+import eu.rssw.pct.elements.DatasetElement;
+import eu.rssw.pct.elements.ElementKind;
+import eu.rssw.pct.elements.EventElement;
+import eu.rssw.pct.elements.MethodElement;
+import eu.rssw.pct.elements.PropertyElement;
+import eu.rssw.pct.elements.QueryElement;
+import eu.rssw.pct.elements.TableElement;
+import eu.rssw.pct.elements.VariableElement;
+
+/**
+ * Import debug segment information from rcode.
+ */
+public class RCodeInfo {
+  // Magic number, followed by same magic number written as little-endian
+  private static final int MAGIC1 = 0x56CED309;
+  private static final int MAGIC2 = 0x09D3CE56;
+
+  // Header values
+  private static final int HEADER_SIZE = 68;
+  private static final int HEADER_OFFSET_MAGIC = 0;
+  private static final int HEADER_OFFSET_TIMESTAMP = 4;
+  private static final int HEADER_OFFSET_MD5 = 10;
+  private static final int HEADER_OFFSET_RCODE_VERSION = 14;
+  private static final int HEADER_OFFSET_SEGMENT_TABLE_SIZE = 0x1E;
+  private static final int HEADER_OFFSET_SIGNATURE_SIZE = 56;
+  private static final int HEADER_OFFSET_TYPEBLOCK_SIZE = 60;
+  private static final int HEADER_OFFSET_RCODE_SIZE = 64;
+
+  // Segment table values
+  private static final int SEGMENT_TABLE_OFFSET_INITIAL_VALUE_SEGMENT_OFFSET = 0;
+  private static final int SEGMENT_TABLE_OFFSET_INITIAL_VALUE_SEGMENT_SIZE = 16;
+  private static final int SEGMENT_TABLE_OFFSET_DEBUG_SEGMENT_OFFSET = 12;
+  private static final int SEGMENT_TABLE_OFFSET_DEBUG_SEGMENT_SIZE = 28;
+
+  private ByteOrder order;
+  private int version;
+  private boolean sixtyFourBits;
+  private long timeStamp;
+  private int md5;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "md5" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1068)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the "md5" field and declare it as a local variable in the relevant methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1450)